ui/src/criteria/attribute.tsx

```diff
@@ -81,73 +136,216 @@ function AttributeEdit(props: AttributeEditProps) {
+                  checked={selectionIndex(hint) > -1}
+                  onChange={() => {
+                    props.dispatchFn(
+                      produce(props.data, (data) => {
+                        if (selectionIndex(hint) > -1) {
+                          data.selected.splice(selectionIndex(hint), 1);
+                        } else {
+                          data.selected.push({
+                            id: hint.attributeValue?.int64Val || -1,
+                            name: hintDisplayName(hint),
+                          });
+                        }
+                      })
+                    );
+                  }}
+                />
+              }
+            />
+          </ListItem>
+        ))}
+      </>
     );
   }
+
   return (
-    <>
-      {enumHintValues?.map((hint: tanagra.EnumHintValue) => (
-        <ListItem key={hintDisplayName(hint)}>
-          <FormControlLabel
-            label={hintDisplayName(hint)}
-            control={
-              <Checkbox
-                size="small"
-                checked={selectionIndex(hint) > -1}
-                onChange={() => {
-                  props.dispatchFn(
-                    produce(props.data, (data) => {
-                      if (selectionIndex(hint) > -1) {
-                        data.selected.splice(selectionIndex(hint), 1);
-                      } else {
-                        data.selected.push({
-                          id: hint.attributeValue?.int64Val || -1,
-                          name: hintDisplayName(hint),
-                        });
-                      }
-                    })
-                  );
-                }}
-              />
-            }
-          />
-        </ListItem>
-      ))}
-    </>
+    <Typography>
+      No information for attribute {props.data.attribute}.
+    </Typography>
   );
 }
 
-type ConceptDetailsProps = {
+type AttributeDetailsProps = {
   data: Data;
 };
 
-function ConceptDetails(props: ConceptDetailsProps) {
-  return (
-    <>
-      {props.data.selected.length === 0 ? (
-        <Typography variant="body1">None selected</Typography>
-      ) : (
-        props.data.selected.map(({ id, name }) => (
+function AttributeDetails(props: AttributeDetailsProps) {
+  if (
+    typeof props.data.selected !== "undefined" &&
+    props.data.selected.length !== 0
+  ) {
+    return (
+      <>
+        {props.data.selected.map(({ id, name }) => (
           <Stack direction="row" alignItems="baseline" key={id}>
             <Typography variant="body1">{id}</Typography>&nbsp;
             <Typography variant="body2">{name}</Typography>
           </Stack>
-        ))
-      )}
+        ))}
+      </>
+    );
+  } else if (isValid(props.data.min) && isValid(props.data.max)) {
+    return (
+      <>
+        <Stack direction="row" alignItems="baseline">
+          <Typography variant="body1">
+            Current {props.data.attribute} in Range {props.data.min} to{" "}
```

Comment:
This should be the name instead of the attribute. For now, in the `registerCriteriaPlugin`  function store the name from the config into a new variable in data. Longer term it might make sense to pull in the name more generically.

Reply:
done